### PR TITLE
Load job UUIDs before checking auth

### PR DESCRIPTION
**Pull request by pschorf**

### CODE REVIEW

File: scheduler/src/cook/mesos/api.clj [hunk 1]
```diff
@@ -870,6 +870,12 @@
        :job/_instance
        :job/uuid))
 
+(defn- wrap-seq [v]
```
Comment (dposada) [inline scheduler/src/cook/mesos/api.clj]: Can we add a docstring?

File: scheduler/test/cook/test/mesos/api.clj [hunk 2]
```diff
@@ -1145,6 +1145,35 @@
...
+          is-authorized-fn (partial auth/is-authorized? {:authorization-fn 'cook.authorization/configfile-admins-auth-open-gets})
+          agent-query-cache (Object.)
+
+          handler (api/destroy-jobs-handler conn framework-id is-authorized-fn agent-query-cache)
+          {:keys [uuid] :as job} (assoc (minimal-job) :user "creator")
+          _ (api/create-jobs! conn {::api/jobs [job]})
+          resp (handler {:request-method :delete
+                         :authorization/user "destroyer"
+                         :query-params {:job uuid}})]
+      (is (= 403 (:status resp))))))
```
Comment (dposada) [inline scheduler/test/cook/test/mesos/api.clj]: Almost all the code is duplicated between the two `testing` clauses, should we extract some of it?

File: scheduler/src/cook/mesos/api.clj (hunk 1 above)
Comment (pschorf) [reply scheduler/src/cook/mesos/api.clj]: Done

File: scheduler/test/cook/test/mesos/api.clj (hunk 2 above)
Comment (pschorf) [reply scheduler/test/cook/test/mesos/api.clj]: Done

**Comment by pschorf** [comment]
Currently, any user can delete any other user's jobs (since ::jobs is nil, no jobs are checked for auth)

**Comment by dposada** [comment]
Should we add an integration test? It would require the multiple user support we discussed a few days ago, right? Is that overkill?

**Comment by pschorf** [comment]
I've tested it out internally and it works, I'd vote to punt on the integration test for now and add it once we add multi-user support.

**Comment by dposada** [comment]
SGTM, let's create an issue for it then

**Pull request merged**